gen/vsup_ifis

```diff
@@ -65,35 +65,43 @@ my $users;  # $users->{$uco}->{ATTR} = $attrValue;
 # AGGREGATE DATA
 #
 # FOR EACH USER
-foreach my $rData ($data->getChildElements) {
+foreach my $resourceId ($data->getResourceIds()) {
 
-	my %resourceAttributes = attributesToHash $rData->getAttributes;
-	my $relationType = $resourceAttributes{$A_R_RELATION_TYPE};
+	my $relationType = $data->getResourceAttributeValue(resource => $resourceId, attrName => $A_R_RELATION_TYPE);
 
 	# Must be in a relation
 	unless ($relationType) {
 		next;
 	}
 
-	my @membersData = $rData->getChildElements;
-
-	foreach my $member (@membersData) {
-
-		my %uAttributes = attributesToHash $member->getAttributes;
-
-		my $key = $uAttributes{$A_UCO};
-
-		$users->{$key}->{$A_TITLE_BEFORE} = $uAttributes{$A_TITLE_BEFORE} || '';
-		$users->{$key}->{$A_FIRST_NAME} = ($uAttributes{$A_ARTISTIC_FIRST_NAME} || ($uAttributes{$A_FIRST_NAME} || ''));
-		$users->{$key}->{$A_LAST_NAME} = ($uAttributes{$A_ARTISTIC_LAST_NAME} || ($uAttributes{$A_LAST_NAME} || ''));
-		$users->{$key}->{$A_TITLE_AFTER} = $uAttributes{$A_TITLE_AFTER} || '';
-
-		$users->{$key}->{$A_BIRTH_NUMBER} = $uAttributes{$A_BIRTH_NUMBER} || '';
-		$users->{$key}->{$A_GENDER} = $uAttributes{$A_GENDER} || '';
-		$users->{$key}->{$A_JOB} = $uAttributes{$A_JOB} || '';
-
-		$users->{$key}->{'EMAIL'} = ($uAttributes{$A_VSUP_PREF_MAIL} || ($uAttributes{$A_VSUP_MAIL} || $uAttributes{$A_LOGIN} . '@vsup.cz'));
-		$users->{$key}->{$A_PHONE} = $uAttributes{$A_PHONE} || '';
+	foreach my $memberId ($data->getMemberIdsForResource(resource => $resourceId)) {
+
+		my $key = $data->getUserAttributeValue(member => $memberId, attrName => $A_UCO);
+		my $titleBefore = $data->getUserAttributeValue(member => $memberId, attrName => $A_TITLE_BEFORE);
+		my $artisticFirstName = $data->getUserAttributeValue(member => $memberId, attrName => $A_ARTISTIC_FIRST_NAME);
+		my $firstName = $data->getUserAttributeValue(member => $memberId, attrName => $A_FIRST_NAME);
+		my $artisticLastName = $data->getUserAttributeValue(member => $memberId, attrName => $A_ARTISTIC_LAST_NAME);
+		my $lastName = $data->getUserAttributeValue(member => $memberId, attrName => $A_LAST_NAME);
+		my $titleAfter = $data->getUserAttributeValue(member => $memberId, attrName => $A_TITLE_AFTER);
+		my $birthNumber = $data->getUserAttributeValue(member => $memberId, attrName => $A_BIRTH_NUMBER);
+		my $gender = $data->getUserAttributeValue(member => $memberId, attrName => $A_GENDER);
+		my $job = $data->getUserAttributeValue(member => $memberId, attrName => $A_JOB);
+		my $prefMail = $data->getUserAttributeValue(member => $memberId, attrName => $A_VSUP_PREF_MAIL);
+		my $vsupMail = $data->getUserAttributeValue(member => $memberId, attrName => $A_VSUP_MAIL);
+		my $userLogin = $data->getUserFacilityAttributeValue(member => $memberId, attrName => $A_LOGIN);
```

Comment:
We use single login-namespace in this service, so the attribute ``login`` is of ``user`` type and not ``user_facility``.

Reply:
Fixed